### DISCUSSION TITLE
[nginx-lua] upstream servers not shown in the topology view in skywalking ui

### DISCUSSION THREAD

**Discussion by hehailong5** [apache/skywalking#9532] [Q&A]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar issues.

### Apache SkyWalking Component

Nginx Lua Agent (apache/skywalking-nginx-lua)

### What happened

I have two openresty well configured according to the example in the github page and  both are connected to the same skywalking apm.

the data flow looks like below:
client ->  openresty1 -> server1 -> openresty2 -> server2

but in the topology view, it shows only:
user -> openresty1 -> openresty2 

### What you expected to happen

in the topology view, I would expect to see:
user -> openresty1 -> server1 -> openresty2 -> server2

### How to reproduce

as described in what happened part

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
First, this should not be submitted as a bug. Because the description is not relative to any feature.

OpenResty to service to another OpenResty2 is an common case, what you described only could say you don't have propagated tracing context through service.

If you want to understand, you should check the propogated header. And read STAM paper and protocol doc in the doc site to understand whether the headers are correct.

If you don't provide details, we don't know why.

**Comment by hehailong5** [comment]
Hi,
Thanks for the quick reply and sorry for the churn.
here are the configuration details:

Openresty1:
lua_shared_dict tracing_buffer 100m;
init_worker_by_lua_block {
        local metadata_buffer = ngx.shared.tracing_buffer
        metadata_buffer:set('serviceName', 'openresty1')
        metadata_buffer:set('serviceInstanceName', 'openresty1-0')
        require("skywalking.util").set_randomseed()
        require("skywalking.client"):startBackendTimer('http://<ip of sw apm>:12800')
        skywalking_tracer = require("skywalking.tracer")
}

log_by_lua_block {
        skywalking_tracer:prepareForReport()
}

access_by_lua_block {
        skywalking_tracer:start('server1')
}

body_filter_by_lua_block {
        if ngx.arg[2] then
              skywalking_tracer:finish()
        end
}

Openresty2:
lua_shared_dict tracing_buffer 100m;
init_worker_by_lua_block {
        local metadata_buffer = ngx.shared.tracing_buffer
        metadata_buffer:set('serviceName', 'openresty2')
        metadata_buffer:set('serviceInstanceName', 'openresty2-0')
        require("skywalking.util").set_randomseed()
        require("skywalking.client"):startBackendTimer('http://<ip of sw apm>:12800')
        skywalking_tracer = require("skywalking.tracer")
}

log_by_lua_block {
        skywalking_tracer:prepareForReport()
}

access_by_lua_block {
        skywalking_tracer:start('server2')
}

body_filter_by_lua_block {
        if ngx.arg[2] then
              skywalking_tracer:finish()
        end
}

server1 (nginx):

    server {
        listen       8081;
        location /test {
                proxy_pass http://<address of openresty2>;
        }
    }

server2 (nginx):

    server {
        listen       8082;
        location /test {
                return "hello from server2";
        }
    }


sorry I have no idea about "propogated header" or "STAM paper and protocol doc" you mentioned here at all.
as you said, this is a very common use case, so please help review the configurations above and kindly correct what I may make wrong or point out what may miss.

Thank you!

**Comment by wu-sheng** [comment]
- https://skywalking.apache.org/docs/main/latest/en/papers/stam/
- https://skywalking.apache.org/docs/main/latest/en/protocols/readme/#tracing

**Comment by Hello-Linux** [comment]
@hehailong5 My situation is exactly the same as yours. I have two nginx. The topology also shows one nginx instead of two nginx。
the data flow looks like below:
client -> openresty1 -> server1 -> openresty2 -> server2

but in the topology view, it shows only:
user -> openresty1 -> server1

**Comment by Hello-Linux** [comment]
@hehailong5 Did you solve this problem?

**Reply by hehailong5** [reply]
No. Actually we turn to zipkin at last, since no one helps on this here :(
